dask_cloudprovider/providers/gcp/instances.py

```diff
@@ -0,0 +1,326 @@
+import asyncio
+import uuid
+
+import dask
+from dask_cloudprovider.providers.generic.vmcluster import (
+    VMCluster,
+    VMInterface,
+    SchedulerMixin,
+    WorkerMixin,
+)
+
+from dask_cloudprovider.utils.socket import is_socket_open
+
+try:
+    import googleapiclient.discovery
+    from googleapiclient.errors import HttpError
+    from google.auth.exceptions import DefaultCredentialsError
+except ImportError as e:
+    msg = (
+        "Dask Cloud Provider GCP requirements are not installed.\n\n"
+        "Please either conda or pip install as follows:\n\n"
+        "  conda install -c conda-forge google-api-python-client  # either conda install\n"
+        '  python -m pip install "dask-cloudprovider[gcp]" --upgrade  # or python -m pip install'
+    )
+    raise ImportError(msg) from e
+
+
+class GCPInstance(VMInterface):
+    def __init__(
+        self,
+        cluster,
+        name,
+        config=None,
+        zone=None,
+        projectid=None,
+        machine_type=None,
+        filesystem_size=None,
+        source_image=None,
+        docker_image=None,
+        ngpus=None,
+        gpu_type=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.cluster = cluster
+        self.name = name
+        self.config = config
+        self.projectid = projectid or self.config.get("projectid")
+        self.zone = zone or self.config.get("zone")
+
+        self.machine_type = machine_type or self.config.get("machine_type")
+
+        self.source_image = source_image or self.config.get("source_image")
+        self.docker_image = docker_image or self.config.get("docker_image")
```

Comment:
If I do not set this via a kwarg or in my config the scheduler fails to start. I think this is because we are overriding the default from `VMInterface` here. We should ensure the default passes through or set the default in the config YAML.

```suggestion
        self.docker_image = docker_image or self.config.get("docker_image", self.docker_image)
```